features/79-structured-logging/api.md

```diff
@@ -0,0 +1,148 @@
+# Structured Logging
+
+## Feature Overview
+
+* Support production logs filtering
+* Show logs with color to make different kinds of log easier to see
+
+## Requirements
+
+* Support different filtering to make developer easier to discover the logs they want
+    1. filter by log level
+        - warning
+        - error
+        - info
+        - debug
+    1. filter by component
+        - Server
+        - Chat Plugin
+        - CMS Plugin
+        - Forgot Password Plugin
+        - SSO Plugin
+        - Cloud Code Functions (Default value of log functions, so user doesn't have to give too many argument when using log function in their code)
+    1. filter by tag
+        - db
+        - request
+        - handler
+        - pubsub
+        - Other with user input (User can defined their own tag if they use SDK provided logging functions to print log)
+    1. All message / System logs or Errors
+        - All message (No filter applied)
+        - System logs or Errors (All logs sent out not in JSON format, including system logs or exceptions. If user use `print` or `console.log` without using SDK log function in their cloud function, those logs will also be shown here)
+
+* A toggle to allow user group logs by request ID (Logs without request ID will not be shown when this is ON)
```

Comment:
Sorry for my ignorance, is request ID something automatically generated by skygear-server no matter what now, so that meant users can look at what logs are generated by API requests?